### PR TITLE
Disallow non-sequential version

### PR DESCRIPTION
**Pull request by oguzhanunlu**

### CODE REVIEW

File: src/main/scala/com/snowplowanalytics/iglu/server/model/VersionCursor.scala [hunk 1]
```diff
@@ -59,8 +62,7 @@ object VersionCursor {
     patchesAllowed: Boolean
   ): Either[Inconsistency, Unit] =
     if (existing.contains(version) && !patchesAllowed) Inconsistency.AlreadyExists.asLeft
```
Comment (istreeter) [inline src/main/scala/com/snowplowanalytics/iglu/server/model/VersionCursor.scala:64]: I _think_ your `isVersionAllowed` does not properly consider the case when the user is trying to patch an existing schema.  I _think_ this line needs to change to something like:

```scala
if (existing.contains(version))
  Either.cond(patchesAllowed, (), Inconsistency.AlreadyExists)
else
  isVersionAllowed(existing, get(version))
```

Please check that thoroughly though!  I haven't thought it through too deeply.

**Review by istreeter — COMMENTED** [review]
Looking good.

File: src/main/scala/com/snowplowanalytics/iglu/server/model/VersionCursor.scala (hunk 1 above)
Comment (oguzhanunlu) [reply src/main/scala/com/snowplowanalytics/iglu/server/model/VersionCursor.scala:64]: I tested all 4 cases of `VersionCursor` and it works as expected, returning `Right(())`

But I like your snippet more, it is easier to understand compared to digging into `isVersionAllowed`

**Review by istreeter — APPROVED** [review]

**Comment by spenes** [comment]
Moved [here](https://github.com/snowplow/iglu-server/pull/146), closing this one.

**Pull request closed without merging**